conduit/pipeline/pipeline_test.go

```diff
@@ -744,15 +767,39 @@ func TestPipelineRetryVariables(t *testing.T) {
 			err := pImpl.Init()
 			assert.Nil(t, err)
 			before := time.Now()
+			done := false
+			// test for "unlimited" timeout
+			go func() {
+				time.Sleep(maxDuration)
+				if !done {
+					cf()
+					assert.Equal(t, testCase.totalDuration, maxDuration)
+				}
+			}()
 			pImpl.Start()
 			pImpl.wg.Wait()
 			after := time.Now()
 			timeTaken := after.Sub(before)
 
-			msg := fmt.Sprintf("seconds taken: %s, expected duration seconds: %s, epsilon: %s", timeTaken.String(), testCase.totalDuration.String(), testCase.epsilon.String())
-			assert.WithinDurationf(t, before.Add(testCase.totalDuration), after, testCase.epsilon, msg)
-			assert.Equal(t, errImporter.GetBlockCount, testCase.retryCount+1)
+			msg := fmt.Sprintf("seconds taken: %s, expected duration seconds: %s, epsilon: %s", timeTaken.String(), testCase.totalDuration.String(), epsilon)
+			assert.WithinDurationf(t, before.Add(testCase.totalDuration), after, epsilon, msg)
+			if testCase.retryCount == 0 {
+				assert.GreaterOrEqual(t, errImporter.GetBlockCount, uint64(1))
+			} else {
+				assert.Equal(t, errImporter.GetBlockCount, testCase.retryCount+1)
+			}
+			done = true
+			for hook := hook.AllEntries(); len(hook) > 0; hook = hook[1:] {
 
+			}
```

Comment:
```suggestion
```
This seems like a no-op. Am I missing something?

Reply:
woops, this was a co-pilot special that I forgot to remove

Reply:
At least we're not a CRISPR shop and this isn't CRISPR Co-Pilot suggesting a human ending virus mutation.